DEVELOPER.md

```diff
@@ -0,0 +1,359 @@
+
+All interactions with the testsys cluster should be done using the model library through `TestManager`, `TestClient`, or `ResourceClient`.
+
+The model provides [TestManager](model/src/test_manager/manager.rs) which can be used as an entry point for user interactions with testsys.
+To see how to use the TestManager look at the [example cli](cli).
+
+Model also contains a [TestClient](model/src/clients/test_client) for simple interactions with `Test`s and a [ResourceClient](model/src/clients/resource_client) for `Resource`s.
+
+### [Selftest](selftest)
+
+Selftest is a library used to validate incoming pull request by testing the overall testsys framework.
+
+## Getting Started
+
+Testsys is currently tested on `x86_64` machines, these steps may not work as expected on other operating systems.
+
+### Setting up testsys
+
+#### Create a testsys directory
+Create a directory that you wish use as a working directory and navigate to it. 
+The example provided creates a directory in the users home, but a directory may be created anywhere.
+
+```bash
+cd ~
+mkdir testsys && cd testsys
+pwd 
+```
+
+#### Clone the testsys repo
+While in the testsys directory, clone the `bottlerocket-test-system` repo from github.
+
+```bash
+git clone https://github.com/bottlerocket-os/bottlerocket-test-system.git
+cd bottlerocket-test-system
+export TESTSYS_DIR=$(pwd)
+```
+
+The testsys directory is now set up.
+
+### Installing tools
+
+#### Install Kind
+Instead of creating a long-standing eks cluster for testsys development, kind allows the quick creation and destruction of k8s clusters which helps improve the testing experience.
+Follow the [installation instructions](https://kind.sigs.k8s.io/docs/user/quick-start/) to set up kind.
+
+#### Install Kubectl
+While all testing user interactions can be done with `testsys`, it is helpful to look at test, resource, and pod objects in a k8s cluster with `kubectl`.
+Follow the [installation instructions]( https://kubernetes.io/docs/tasks/tools/) to set up kubectl.
+
+#### Install Docker
+Docker is required for building testsys images if docker is not installed follow the [installation instructions](https://docs.docker.com/get-docker/).
+
+## Running a test
+
+### Navigate to the testsys directory
+```bash
+cd ${TESTSYS_DIR}
+```
+
+### Build testing images
+First, the images need to be created and tagged so they can be loaded into a kind cluster.
+```bash
+make controller
+make example-test-agent
+docker tag controller controller:demo
+docker tag example-test-agent example-test-agent:demo
+```
+This step uses the Makefile targets for building the controller image and the example test agent image. The images then need to be tagged so that the cluster uses loaded images instead of ones from external sources.
+
+### Install the testsys cli
+Install the `cli` from the repo.
+```bash
+cargo install --path cli
```

Comment:
This installs the testsys CLI as literally `cli` in my cargo bin:

```
❯ which cli
~/.cargo/bin/cli
❯ cli help
cli 0.0.1
The command line interface for setting up a Bottlerocket TestSys cluster and running tests
```

I'm not really seeing a way for `cargo install` to name the crate something different. Maybe we can look at naming this something more sensible in a different issue? Sounds like this CLI will also soon be moving to the core bottlerocket repo?

Reply:
The testsys cli is moving to the bottlerocket repo. Cli was created as an example cli for anyone wanting to use the testsys framework.

Reply:
Good to know! Probably worth having discussing outside of this pull request. But thanks for the context!